### ISSUE TITLE
Odd batch loading behaviour - spiky 'fraction_of_n_full'

### ISSUE THREAD

**Issue by RustyPanda** [RustyPanda/zoobot#21]
<img width="364" alt="screen shot 2018-06-26 at 13 11 03" src="https://user-images.githubusercontent.com/7740526/41910730-725ee940-7942-11e8-93f5-23653f343532.png">

<img width="365" alt="screen shot 2018-06-26 at 13 12 15" src="https://user-images.githubusercontent.com/7740526/41910770-915ef95c-7942-11e8-8a33-cde42a644cf8.png">

Stratified sample includes a mysterious debug logger that shows very odd behaviour - sharp semi-periodic spikes. I should find out what's going on! 

Perhaps it's related to the start or end of a batch, or a variable batch size.

The names are fraction_of_32_full and fraction_of_12800_full. 100 * batch size of 128?

Periodic, but only sometimes: 
For augs_both, happens every 600 steps
For augs, happens at 200, 1.05, 1.6, then stable

May also be related to the varied training loss and tensorboard silent keyerrors.

**Comment by RustyPanda** [comment]
I added a tensorboard summary measuring the length of batches. They are always 128 (fixed size). So this is not measuring batch length.

For the next run, I've added a tensorboard summary for the mean label per batch.
